registry/handlers/manifests.go

```diff
@@ -156,18 +160,25 @@ func (imh *manifestHandler) GetManifest(w http.ResponseWriter, r *http.Request)
 	manifestList, isManifestList := manifest.(*manifestlist.DeserializedManifestList)
 	if isSchema2 {
 		manifestType = manifestSchema2
-	} else if _, isOCImanifest := manifest.(*ocischema.DeserializedManifest); isOCImanifest {
-		manifestType = ociSchema
+	} else if ociManifest, isOCImanifest := manifest.(*ocischema.DeserializedManifest); isOCImanifest {
+		if ociManifest.MediaType == v1.MediaTypeImageManifest {
+			manifestType = ociImageSchema
+		} else if ociManifest.MediaType == v1.MediaTypeArtifactManifest {
+			manifestType = ociArtifactSchema
+		}
```

Comment:
nit: Do we want to refactor these using switch-case?

Reply:
I'll do this after I have tested everything.